Source/missiles.cpp

```diff
@@ -1466,10 +1466,10 @@ void __fastcall CheckMissileCol(int i, int mindam, int maxdam, bool shift, int m
 			}
 			goto LABEL_13;
 		}
-		if (monster[v10]._mFlags & 0x10) {
-			v18 = dMonster[0][my + 112 * mx];
+		if (monster[v10]._mFlags & MFLAG_TARGETS_MONSTER) {
+			v18 = dMonster[mx][my];
 			if (v18 > 0) {
-				if (monster[v18 - 1]._mFlags & 0x20) /* fix */
+				if (monster[v18 - 1]._mFlags & MFLAG_GOLEM) /* fix */

```

Comment:
What is the fix comment referring to?

Reply:
No clue. @galaxyhaxz ?

Reply:
Anytime you see a block comment with `fix/check` it means the code should be double checked in IDA to make sure it's the right offset. I didn't know about IDA's target delta feature then so many arrays with subtraction were decompiled wrong and they may not be correct.

Reply:
Added a more descriptive note, hope that works:
https://github.com/diasurgical/devilution/commit/a8e5319838ed53506e4f6cfd533feb36809fa4c4